src/lex.cpp

```diff
@@ -628,18 +632,22 @@ void lexer::parse_binary_number() {
     input += 1;
   }
 
+  report_garbage_in_number_literal(input);
+}
+
+void lexer::report_garbage_in_number_literal(const char8* input) {
```

Comment:
Thanks for making a function! It's much easier to read `lexer::parse_binary_number` now.